cpp/src/parquet/statistics.cc

```diff
@@ -620,7 +626,11 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
  private:
   const ColumnDescriptor* descr_;
   bool has_min_max_ = false;
+  // Currently, has_null_count_ is always true, and would
+  // be collected and encoded to `EncodedStatistics`.
   bool has_null_count_ = false;
+  // Currently, has_distinct_count_ would not be encoded
```

Comment:
It seems that `statistics_` already has a set of `hasXXX` variables. Can we reuse those directly?

```cpp
class PARQUET_EXPORT EncodedStatistics {
  std::shared_ptr<std::string> max_, min_;
  bool is_signed_ = false;

 public:
  EncodedStatistics()
      : max_(std::make_shared<std::string>()), min_(std::make_shared<std::string>()) {}


  int64_t null_count = 0;
  int64_t distinct_count = 0;

  bool has_min = false;
  bool has_max = false;
  bool has_null_count = false;
  bool has_distinct_count = false;
```